bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200GatewayHandler.java

```diff
@@ -405,35 +440,36 @@ private void updateChildren(Map<Channel, JsonObject> sendMap, KM200GatewayHandle
                     String tmpService = KM200Utils.checkParameterReplacement(actChannel, remoteDevice);
-                    if (parent == null || parent.equals(remoteDevice.getServiceObject(tmpService).getParent())) {
-                        synchronized (sendMap) {
-                            if (sendMap.containsKey(actChannel)) {
-                                state = dataHandler.parseJSONData(sendMap.get(actChannel),
-                                        remoteDevice.getServiceObject(tmpService).getServiceType(), tmpService,
-                                        actChannel.getAcceptedItemType(),
-                                        KM200Utils.getChannelConfigurationStrings(actChannel));
-                            } else {
-                                state = dataHandler.getProvidersState(tmpService, actChannel.getAcceptedItemType(),
-                                        KM200Utils.getChannelConfigurationStrings(actChannel));
+                    KM200ServiceObject tmpSerObjekt = remoteDevice.getServiceObject(tmpService);
+                    if (null != tmpSerObjekt) {
+                        if (parent == null || parent.equals(tmpSerObjekt.getParent())) {
+                            synchronized (sendMap) {
+                                if (sendMap.containsKey(actChannel)) {
+                                    state = dataHandler.parseJSONData(sendMap.get(actChannel),
+                                            tmpSerObjekt.getServiceType(), tmpService, actChTypes,
+                                            KM200Utils.getChannelConfigurationStrings(actChannel));
+                                } else {
+                                    state = dataHandler.getProvidersState(tmpService, actChTypes,
+                                            KM200Utils.getChannelConfigurationStrings(actChannel));
+                                }
                             }
-                        }
-                        if (state != null) {
-                            try {
-                                gatewayHandler.updateState(actChannel.getUID(), state);
-                            } catch (IllegalStateException e) {
-                                logger.error("Could not get updated item state", e);
+                            if (state != null) {
+                                try {
+                                    gatewayHandler.updateState(actChannel.getUID(), state);
+                                } catch (IllegalStateException e) {
+                                    logger.error("Could not get updated item state", e);
+                                }
                             }
                         }
-                    }
-                    try {
-                        Thread.sleep(readDelay);
-                    } catch (InterruptedException e) {
-                        continue;
+                        try {
+                            Thread.sleep(readDelay);
```

Comment:
Why are you sleeping in a synchronized block? Or better question, why do you need to sleep the thread?

Reply:
The device needs a bit time to recover.